pennylane/optimize/momentum.py

```diff
@@ -55,20 +56,44 @@ def apply_grad(self, grad, x):
         Returns:
             array: the new values :math:`x^{(t+1)}`
         """
-
-        grad_flat = _flatten(grad)
-        x_flat = _flatten(x)
+        args_new = list(args)
 
         if self.accumulation is None:
-            self.accumulation = [self._stepsize * g for g in grad_flat]
-        else:
-            self.accumulation = [
-                self.momentum * a + self._stepsize * g for a, g in zip(self.accumulation, grad_flat)
-            ]
+            self.accumulation = [None] * len(args)
+
+        trained_index = 0
+        for index, arg in enumerate(args):
+            if getattr(arg, "requires_grad", True):
+                x_flat = _flatten(arg)
+                grad_flat = _flatten(grad[trained_index])
+                trained_index += 1
+
+                self._update_momentum(index, grad_flat)
+
+                x_new_flat = [e - a for a, e in zip(self.accumulation[index], x_flat)]
+
+                args_new[index] = unflatten(x_new_flat, arg)
 
-        x_new_flat = [e - a for a, e in zip(self.accumulation, x_flat)]
+                if isinstance(arg, ndarray):
+                    args_new[index] = args_new[index].view(tensor)
+                    args_new[index].requires_grad = True
 
-        return unflatten(x_new_flat, x)
+        return args_new
+
+    def _update_momentum(self, index, grad_flat):
+        r"""Update the momentum
+
+        Args:
+            index (Int): location of arg to update
+            grad_flat (list): flattened list form of gradient
+        """
```

Comment:
I like that you've factored this into its own method, feels a lot cleaner
```suggestion
    def _update_momentum(self, index, grad_flat):
        r"""Update the momentum.

        Args:
            index (int): index of argument to update
            grad_flat (list): flattened list form of gradient
        """
```